build.rs

```diff
@@ -71,8 +77,11 @@ fn write_timezone_file(timezone_file: &mut File, table: &Table) {
 
     write!(timezone_file,
 "impl FromStr for Tz {{
+    #[cfg(feature = \"std\")]
     type Err = String;
-    fn from_str(s: &str) -> Result<Self, String> {{
+    #[cfg(not(feature = \"std\"))]
+    type Err = &'static str;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {{
```

Comment:
I wonder if we should just introduce a better error struct here, I don't love the idea of the no_std and std features being different, but we could bump a version and provide better errors overall.

Reply:
I'm still figuring out how to design good error types in some libraries I'm working on, so I'll leave that up to you.

Reply:
great, yeah I can handle that!